source/common/modules/markdown-editor/util/remove-markdown-link.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        removeMarkdownLink function
+ * CVM-Role:        Utility function
+ * Maintainer:      Hendrik Erz
+ * License:         GNU GPL v3
+ *
+ * Description:     This function removes a markdown link, performing necessary
+ *                  transformations where applicable.
+ *
+ * END HEADER
+ */
+
+import { type EditorView } from '@codemirror/view'
+import { type ChangeSpec } from '@codemirror/state'
+import { type SyntaxNode } from '@lezer/common'
+
+/**
+ * Utility to remove a markdown link. It extracts the text from the markdown
+ * link and replaces the link with just the text.
+ *
+ * @param {EditorView} view - The editor view
+ * @param {SyntaxNode} node - The node containing the Link
+ */
+export function removeMarkdownLink (node: SyntaxNode, view: EditorView): void {
+  // To deal with user being able to click on either part of the link (Text or URL)
+  // And the two different types of links []() and <>
```

Comment:
Reminds me that there's a third type: Reference style: https://spec.commonmark.org/0.31.2/#link-reference-definitions

I am not exactly sure how Codemirror handles these, but with a bunch of console.logs that should be researchable

Reply:
I will look into this too.

Reply:
![image](https://github.com/Zettlr/Zettlr/assets/84065477/58b57f9c-8c8d-4fe7-a6dc-f4a726be2f74)

At the moment the right click context menu doesn't appear when you right click on the [text][label] part of the reference style links. Is this something you want us to implement too?

Reply:
It appears that a bug may be preventing the context menu from displaying for [Text 1][link1]. Additionally, the "remove link" option is currently visible for [link1]: https://example.com/. Nathan, do you have any suggestions on how we should handle this?

From my perspective, once the context menu bug for [Text 1][link1] is resolved, the "remove link" functionality should operate correctly and I think we should possibly raise an issue about this (which we will resolve next). However, the option to remove the link for [link1]: https://example.com/ will still be available. Could you please clarify how you would like us to proceed with this issue?

Reply:
It appears that a bug due to the getURLForNode function call in the linkImageMenu may be preventing the context menu from displaying for [Text 1][link1] as it throws an error because of it not being able to find a URL child. Additionally, the "remove link" option is currently visible for [link1]: https://example.com/. Nathan, do you have any suggestions on how we should handle this?

From my perspective, once the context menu bug for [Text 1][link1] is resolved, the "remove link" functionality should operate correctly and I think we should possibly raise an issue about this (which we will resolve next). However, the option to remove the link for [link1]: https://example.com/ will still be available. Could you please clarify how you would like us to proceed with this issue?

Reply:
I think those references … mh, it seems like we'd have to actually go searching in the syntax tree for the matching references before being able to decide whether to actually show a context menu. I would say this is out of scope for this specific issue, and should be tackled separately, especially since it needs to be fixed within a set of functionality, because iirc I never implemented reference links at any point …

Reply:
Would you like us to report an issue for it?

Reply:
Yes please do so!

Reply:
Yes, done #5142